### PR TITLE
[feature][az] create shopping cart modal

### PR DESCRIPTION
**Pull request by lssndrrr**
here's the current progress:

image

added test data to constant/merch/shoppingCart.ts

have yet to make it responsive on small screens and still does not add the total of items checked. should i move the shoppingCartItem to interface/merch.ts?

### CODE REVIEW

File: components/merch/modals/ShoppingCartItem.tsx [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+interface ShoppingCartItem {
+    merch: MerchItem
+    quantity: number
+    size: MerchSize
+    color: Color
+    maxQuantity: number
+    isCheckedOut: boolean
+}
+
+const ShoppingCartItem = (props: ShoppingCartItem) => {
```
Comment (maxellmilay) [inline components/merch/modals/ShoppingCartItem.tsx:16]: @lssndrrr destruction the props. Maybe something like this:

`const { merch, quantity,... } = props;`

Comment (maxellmilay) [reply components/merch/modals/ShoppingCartItem.tsx:16]: aside from destructuring the merch itself, also destruct the cart orders

File: .gitignore [hunk 2]
```diff
@@ -19,6 +19,8 @@
 # misc
 .DS_Store
 *.pem
+.vs/
+tailwind.extension.json
```
Comment (maxellmilay) [inline .gitignore:23]: @lssndrrr may I ask what is this for?

File: components/merch/modals/ShoppingCartItem.tsx [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    const handleToggle = () => {
+        setIsChecked(prev => !prev)
+    }
+
+    const handleDecrement = () => {
+        setQuantity((prevQuantity) => Math.max(minQuantity, prevQuantity - 1));
+    };
+
+    const handleIncrement = () => {
+        setQuantity((prevQuantity) => Math.min(props.maxQuantity, prevQuantity + 1));
```
Comment (maxellmilay) [inline components/merch/modals/ShoppingCartItem.tsx:42]: @lssndrrr I'd to know more about this explanation. Can you explain it for me in this thread?

File: components/merch/modals/ShoppingCartItem.tsx [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+    };
+
+    return (
+        // button
```
Comment (maxellmilay) [inline components/merch/modals/ShoppingCartItem.tsx:46]: remove comment

File: components/merch/modals/ShoppingCartModal.tsx [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+                    <button className="ml-auto bg-[#b53629] pl-2 pr-2" onClick={handleCheckout}>CHECKOUT</button> 
+                </div>
+            </div>
+        </div>
+    )
+}
+
+export default ShoppingCartModal
+
+    //< button onClick = { handleClose } className = "bg-csg-green-100 h-fit px-1 py-0.5 sm:px-2 sm:py-1 rounded-lg flex flex-row gap-0.5 sm:gap-1 items-center cursor-pointer hover:text-white/75" > {< BsArrowLeft />} CONTINUE SHOPPING</button >
```
Comment (maxellmilay) [inline components/merch/modals/ShoppingCartModal.tsx:70]: @lssndrrr remove comment

**Comment by maxellmilay** [comment]
@lssndrrr kindly add newlines at the end of every file

**Comment by maxellmilay** [comment]
@lssndrrr also update this branch from develop

File: .gitignore (hunk 2 above)
Comment (lssndrrr) [reply .gitignore:23]: Utilized the visual studio extension adding intellisense to tailwind, so i needed to do this. did not work though so will remove it :)

File: components/merch/modals/ShoppingCartItem.tsx (hunk 3 above)
Comment (lssndrrr) [reply components/merch/modals/ShoppingCartItem.tsx:42]: @maxellmilay sorry for the late response! it basically just makes sure na it doesn't go over the max quantity of the item (this is for items na on hand) but if everything sa merch page is preordered then this is not needed :)

**Comment by lssndrrr** [comment]
a bit of a late update bc i got busy with school but i have (mostly) adjusted it to fit smaller screens. just need some feedback on the design since i could not get it to fit with the bar up top so i decided to remove it

image

i have a few more bugs to squash and the delete item and check all to implement but that's about it! 

let me know your thoughts!

ps checkbox for all is not yet aligned so ignore that pls

**Review by maxellmilay — APPROVED** [review]

**Pull request merged**